autoload/lsp/utils/text_edit.vim

```diff
@@ -167,7 +171,7 @@ function! s:generate_sub_cmd_insert(text_edit) abort
     let l:sub_cmd = s:preprocess_cmd(a:text_edit['range'])
     let l:sub_cmd .= s:generate_move_start_cmd(l:start_line, l:start_character)
 
-    if l:start_character >= len(getline(l:start_line))
+    if l:start_character >= len(getline(l:start_line)) && getline(l:start_line) != ''
```

Comment:
```vim
let l:line = getline(l:start_line)
if l:line != '' && l:start_character >= len(l:line)
```
And this check with `l:start_character` must be utf-16 aware.

Reply:
What do you mean? An empty line is an empty line. Why would it need encoding check. As you notice, I've just added another check. Have you tried to execute the `LspCodeAction` on the example I've provided? I understand if it's not an empty line to perform the check of the `l:start_character >= len(getline(l:start_line))`. but it doesn't make sense when we are talking about empty lines. 

Or maybe I don't understand what you are saying. But my only addition is to check for an empty line. It has nothing to do with encodings.

Reply:
l:start_character should be offset based on a UTF-16 string representation. and len(x) return count of bytes.

Reply:
I think mattn was a) asking you to update the _old_ check (similarly to what other recent PRs have done elsewhere; here you'd change `len` to `strchars`) while you're touching that line and b) suggesting a refactor to avoid calling `getline` twice and putting the cheap check first.

@mattn machakann has a branch where he is working on fixing the utf-16 on a wider scale, including this line; but fixing it now makes merging easier, probably.

Reply:
@clason 

Ok. I got it. I thought he was referring to the check I've added. 

Thank you. I will update the PR.